.github/workflows/build-and-test-self-hosted.yaml

```diff
@@ -6,7 +6,15 @@ on:
   workflow_dispatch:
 
 jobs:
+  prevent-running-on-forks:
+    name: Prevent running on forks
+    runs-on: ubuntu-latest
+    if: ${{ (github.event_name != 'schedule') || (github.event_name == 'schedule' && github.repository_owner == 'autowarefoundation') }}
+    steps:
+      - run: "true"
+
   build-and-test-self-hosted:
+    needs: prevent-running-on-forks
```

Comment:
Could you skip `sync-files.yaml` as well? It requires a key of GitHub App and normally does it does not need to be executed.

Reply:
Added: 21d80d974

Reply:
I'm sorry but I've changed my mind. Since `sync-files.yaml` is synced to many repositories now, I guess it's good not to add `prevent-running-on-forks`.